src/extractor/guidance/intersection_generator.cpp

```diff
@@ -44,197 +45,98 @@ Intersection IntersectionGenerator::operator()(const NodeID from_node, const Edg
-        }
         intersection.push_back(
-            ConnectedRoad(TurnOperation{onto_edge,
-                                        angle,
+            ConnectedRoad(TurnOperation{edge_connected_to_intersection,
+                                        0,
                                         bearing,
                                         {TurnType::Invalid, DirectionModifier::UTurn},
                                         INVALID_LANE_DATAID},
-                          turn_is_valid,
-                          out_segment_length));
+                          false));
     }
 
-    // We hit the case of a street leading into nothing-ness. Since the code here assumes
-    // that this
-    // will never happen we add an artificial invalid uturn in this case.
-    if (!has_uturn_edge)
-    {
-        const double bearing =
-            util::coordinate_calculation::bearing(turn_coordinate, first_coordinate);
-
-        intersection.push_back({TurnOperation{via_eid,
-                                              0.,
-                                              bearing,
-                                              {TurnType::Invalid, DirectionModifier::UTurn},
-                                              INVALID_LANE_DATAID},
-                                false,
-                                in_segment_length});
-    }
-
-    std::sort(std::begin(intersection),
-              std::end(intersection),
-              std::mem_fn(&ConnectedRoad::compareByAngle));
-
-    BOOST_ASSERT(intersection[0].angle >= 0. &&
-                 intersection[0].angle < std::numeric_limits<double>::epsilon());
-
-    const auto valid_count =
-        boost::count_if(intersection, [](const ConnectedRoad &road) { return road.entry_allowed; });
-    if (0 == valid_count && uturn_could_be_valid)
+    if (!intersection.empty())
     {
-        // after intersections sorting by angles, find the u-turn with (from_node ==
-        // to_node)
-        // that was inserted together with setting uturn_could_be_valid flag
-        std::size_t self_u_turn = 0;
-        while (self_u_turn < intersection.size() &&
-               intersection[self_u_turn].angle < std::numeric_limits<double>::epsilon() &&
-               from_node != node_based_graph.GetTarget(intersection[self_u_turn].eid))
-        {
-            ++self_u_turn;
-        }
-
-        BOOST_ASSERT(from_node == node_based_graph.GetTarget(intersection[self_u_turn].eid));
-        intersection[self_u_turn].entry_allowed = true;
+        const auto base_bearing = util::bearing::reverseBearing(intersection.begin()->bearing);
+        std::sort(intersection.begin(),
+                  intersection.end(),
+                  [base_bearing](const ConnectedRoad &lhs, const ConnectedRoad &rhs) {
+                      return util::bearing::angleBetweenBearings(base_bearing, lhs.bearing) <
+                             util::bearing::angleBetweenBearings(base_bearing, rhs.bearing);
```

Comment:
I _think_ we already have a comparator which does this (?)

If not it could make sense to provide one in a central place with a default `std::less<>` binop.